packages/iTwinUI-react/src/core/Dialog/Dialog.tsx

```diff
@@ -65,7 +68,9 @@ export const Dialog = Object.assign(
           trapFocus,
         }}
       >
-        {children}
+        <div style={{ isolation: 'isolate', ...style }} {...rest}>
```

Comment:
I think this also needs a z-index of around `999`, because the z-indices of the backdrop and dialog are isolated inside this container.

It can be hardcoded for now but should be moved to CSS later.

Reply:
When I tested, I had some content near the button with `z-index: 1000` and it didn't bleed trough, so maybe `z-index` on this div is not needed.

Reply:
In my testing I found it's broken even with my suggestion of adding z-index to this div, because it doesn't even have a size currently. I think this div will also need `position: fixed` or `position: absolute` and a size or `inset`.

Quick video demonstration:
https://user-images.githubusercontent.com/9084735/187711177-b2e35f09-322c-43df-a771-424683e6a2a6.mp4

Reply:
Removed everything, seems to be working fine.